specification/billing/resource-manager/Microsoft.Billing/preview/2018-03-01-preview/billing.json

```diff
@@ -200,7 +200,47 @@
         }
       }
     },
-    "/subscriptions/{subscriptionId}/providers/Microsoft.Billing/invoices": {
+    "/providers/Microsoft.Billing/discoverTenants?billingProfileId={billingProfileId}": {
```

Comment:
As billingProfileId is mandatory, this is likely better served by a POST action request where the billingProfileId is passed in the request body. See this for an [example ](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/proxy-api-reference.md#subscription-wide-reads-and-actions)

Reply:
This API is designed for Power BI custom connector to achieve multi-tenancy.  This design is approved by connector Team.  It is a request flow which brings Tenant Id for a specified billingProfileId input. API is doing a tenant discovery and that's why it is named as "DiscoverTenant". I hope this clarifies your question.

Reply:
Changes in swagger are committed to making the URL compliant with the Resource Provider Contract. The new URL is
'/providers/Microsoft.Billing/billingProfiles/{billingProfileId}/discoverTenants'

I am keeping the name discoverTenants as the sole purpose of this API is to discover the Tenants only. As all the review comments are implemented can you please mark this approve.

Reply:
thank you for these changes. This path is syntactically correct. The resource type name in this case would be "discoverTenants" which is a bit odd, as verbs are atypical in a resource type name. I would prefer it to be named with a noun pattern for consistency. `discoveredTenants` would be substantially preferable in my opinion.